src/python/pants/backend/go/goals/debug_goals.py

```diff
@@ -193,9 +193,9 @@ async def go_export_cgo_codegen(
     workspace: Workspace,
     golang_subsystem: GolangSubsystem,
 ) -> GoExportCgoCodegen:
-    if not golang_subsystem.cgo_enabled:
+    if not golang_subsystem.cgo_allowed:
         raise ValueError(
-            "Nothing to export since cgo is disabled, which is set by the option [golang].cgo_enabled"
+            "Nothing to export since cgo is disabled, which is set by the option [golang].cgo_allowed"
```

Comment:
Seems odd to still use the word "disabled" in the error message, but it makes more sense than "disallowed", which leads me to believe that perhaps `--cgo-enabled` is an appropriate name after all?

Can you explain the need for this change? 

Also, why would one ever disable this option?

Reply:
To clarify: enabling this by default makes sense, but the "enabled" name seems to already convey that "this option permits cgo builds to occur if a package makes use of cgo support." no? What am I missing?

Reply:
Context:

- In `go` toolchain world, you can set `CGO_ENABLED=0` and prevent any cgo processing from occurring in any transitive dependency. (Bazel rules_go calls this "pure" mode, as in pure Go.) Packages can use the presence or absence of the `cgo` build tag to handle this gracefully.

- For Pants, I would like to add a `cgo_allowed` field to `go_binary` so a user can set `cgo_allowed=False` on a `go_binary` and Pants will disable cgo for all transitive dependencies of that binary. That is https://github.com/pantsbuild/pants/pull/16843 which fails with rule graph errors currently.

This PR extracts the option name change from https://github.com/pantsbuild/pants/pull/16843.

To me, the word "enable" connotes that something _is_ switched on, whereas "allowed" connotes that possibility that something _may_ be switched on. The latter is what this option is intended to mean.

Reply:
I see, thanks, that is helpful. 

But I still note that in the code comments and in your explanation here you use "disable" to explain what happens when `cgo_allowed=False`... So it does seem like "enable"/"disable" is the verbiage that the mind reaches for here?